arp-scan: update to 1.9.6

defanator opened the pull request:
Signed-off-by: Andrei Belov <defanator@gmail.com>

Maintainer: @urusha 
Compile tested: ar71xx / Archer C7 / OpenWrt 18.06.4
Run tested: ar71xx / Archer C7 / OpenWrt 18.06.4

Previous versions do not work with libpcap 1.9.1, see these reports for details:
https://github.com/royhills/arp-scan/issues/42
https://github.com/royhills/arp-scan/issues/2

feckert commented on net/arp-scan/Makefile:
```diff
@@ -41,6 +41,10 @@ endef
+	$(INSTALL_DIR) $(1)/usr/share/arp-scan
+	$(INSTALL_DATA) $(PKG_BUILD_DIR)/ieee-iab.txt $(1)/usr/share/arp-scan/
+	$(INSTALL_DATA) $(PKG_BUILD_DIR)/ieee-oui.txt $(1)/usr/share/arp-scan/
+	$(INSTALL_DATA) $(PKG_BUILD_DIR)/mac-vendor.txt $(1)/usr/share/arp-scan/
```
Make´s this sense to add this additional files?
They are quit big for a small device.
ieee-iab.txt 137kB 
ieee-oui.txt 783kB
I think we should move this files to a separate package.

defanator replied:
@feckert the essential functionality of `arp-scan` tool is in immediate indication of a vendor/manufacturer for each discovered MAC; without this information an output is almost useless. I agree that the database data is quite large for small devices. I see two possible approaches: 1) make separate package with the database, 2) add a couple of scripts - https://github.com/royhills/arp-scan/blob/master/get-iab and https://github.com/royhills/arp-scan/blob/master/get-oui to the base package and provide a note (does `opkg` have something like `post-install` scripts?..) containing the instructions on how to fetch fresh MAC database data to the device using those scripts.

What do you think?

Another possible extension would be to allow `arp-scan` to use gzipped data, but this definitely is separate topic. :)

defanator replied:
Uhm, those updater scripts are written in perl, and I suppose there's no any sense in adding a perl dependency to the arp-scan package... Perhaps separate package with MAC database is the best option as of now. @urusha any thoughts?

urusha commented on net/arp-scan/Makefile:
```diff
@@ -43,4 +43,40 @@ define Package/arp-scan/install
+  TITLE:=MAC database for ARP scanner
+  DEPENDS:=+arp-scan
+  URL:=https://github.com/royhills/arp-scan
+  PKG_MAINTAINER:=Sergey Urushkin <urusha.v1.0@gmail.com>
```
We might add `define Package/arp-scan/default` and use `$(Package/arp-scan/default)` in a particular package. 
Also, `PKG_MAINTAINER` seems to be redundant here.
And arp-scan-database actually doesn't depend on arp-scan, it doesn't even require fixup/configuring/building, just copying, but I'm not sure if we should skip this for a single package (and have such ability). @feckert please advice, what is a better way to accomplish this?

feckert replied:
- This would be good to move this to `define Package/arp-scan/default`
- Also`PKG_MAINTAINER` should be remove from the `Package/arp-scan/` definition
- I think the dependency is not necessary.

neheb commented on net/arp-scan/Makefile:
```diff
@@ -8,12 +8,12 @@
+PKG_HASH:=971b45c3369816467994797fbcd0076eb8f0ffb9c42764ea6dba25ab3fd490da
 
 PKG_MAINTAINER:=Sergey Urushkin <urusha.v1.0@gmail.com>
 PKG_LICENSE:=GPL-3.0
```
Replace with GPL-3.0-or-later

neheb commented on net/arp-scan/Makefile:
```diff
@@ -8,12 +8,12 @@
 PKG_RELEASE:=1
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
 PKG_SOURCE_URL:=https://codeload.github.com/royhills/arp-scan/tar.gz/$(PKG_VERSION)?
```
An alternative is:

```
PKG_SOURCE_URL:=https://github.com/royhills/arp-scan/archive/$(PKG_VERSION)
```

Note the lack of ? as it's not needed.

feckert commented:
@defanator @urusha I don't see any changes yet to be made.
Is that now ready to merge?
If so please stash your commit.
So we have an update commit for arp-scan and a commit thats add the new package arp-scan-database.

defanator commented:
@feckert what do you mean? I've pushed three changes since the last review:

https://github.com/openwrt/packages/pull/10379/commits/d191951026a4c6ae94a4019815f65298d6a92ba0
https://github.com/openwrt/packages/pull/10379/commits/ffb4038a3928beb3e401fc361d98af3a8fe19101
https://github.com/openwrt/packages/pull/10379/commits/fa0443d3762948df45877861c24049f26305b27c

So, all concerns were addressed, except the PKG_SOURCE_URL as it seems to be totally unrelated to the subject of this PR (see comments above).

Let me know if there's anything else I should do.

feckert commented:
I think you should clean that up.
* we have updated the version
* cleaned up the Makefile
* added a new package (arp-scan-database)
But we have 8 commits for this change.

defanator commented:
@feckert could we please just use "squash" merge here?

Please clarify if you want me to open a new PR. I don't quite understand how to re-do 8 commits in 3 in the single branch in git.

feckert commented:
Keyword would be ```git rebase -i HEAD~8```
then do your changes and stash the commits
and then do a ```git push --force``` to update your pullrequst.
But be careful wit --force
Please have a look and read the git docu.
But I have already made the changes in my staging.
Will push this maybe in the next days if I have a time slot if not able to do this.
Thanks for your work

defanator commented:
@feckert thanks for the directions, I _think_ I sorted this out. Now this PR should consist of 2 changes - 1) version (&& license) update, 2) introducing standalone package with MAC databases.

Let me know whether everything is ok now.

feckert commented:
@defanator looks good to me now
@urusha thump up from your side?
If there are no more objections, I will merge that tomorrow.

urusha commented:
Seems good. Thanks @defanator

feckert commented:
@defanator thanks for your contribution
